File: javascript/stage-2/level-3/solution.js
// Algorithm 1: Complex Arithmetic Expression
let a = 10;
let b = 5;
let c = 2;

let result = (a + b) * c - a / c;
console.log("Result: " + result);

let result2 = a + b * c - a / c;
console.log("Result without parentheses: " + result2);

let result3 = ((a + b) * c - a) / c;
console.log("Result with different grouping: " + result3);

console.log(); // Add some spacing

// Algorithm 2: Quadratic Formula Calculator
let a2 = 1;  // Renamed to avoid conflict
let b2 = -5;
let c2 = 6;

let discriminant = b2 * b2 - 4 * a2 * c2;
let sqrt_discriminant = Math.sqrt(discriminant);
let root1 = (-b2 + sqrt_discriminant) / (2 * a2);
let root2 = (-b2 - sqrt_discriminant) / (2 * a2);

console.log("Root 1: " + root1);
console.log("Root 2: " + root2);

console.log(); // Add some spacing

// Algorithm 3: Compound Interest Calculator
let principal = 1000;
let rate = 0.05;
let time = 10;
let compounds_per_year = 12;

let amount = principal * Math.pow(1 + rate / compounds_per_year, compounds_per_year * time);
let interest = amount - principal;

console.log("Principal: $" + principal);
console.log("Final Amount: $" + amount.toFixed(2));
console.log("Interest Earned: $" + interest.toFixed(2));

console.log(); // Add some spacing

// Algorithm 4: Geometric Calculations
let radius = 5;
let side = 4;
let base = 6;
let height = 8;

let circle_area = Math.PI * radius * radius;
let circle_circumference = 2 * Math.PI * radius;
let square_area = side * side;
let square_perimeter = 4 * side;
let triangle_area = 0.5 * base * height;

console.log("Circle - Area: " + circle_area.toFixed(2) + ", Circumference: " + circle_circumference.toFixed(2));
console.log("Square - Area: " + square_area + ", Perimeter: " + square_perimeter);
console.log("Triangle - Area: " + triangle_area);

console.log(); // Add some spacing

// Algorithm 5: Physics Formula Calculator
let mass = 5.5;
let velocity = 10;
let acceleration = 9.8;
let time2 = 3;  // Renamed to avoid conflict

let kinetic_energy = 0.5 * mass * velocity * velocity;
let force = mass * acceleration;
let distance = velocity * time2 + 0.5 * acceleration * time2 * time2;
let momentum = mass * velocity;

console.log("Kinetic Energy: " + kinetic_energy);
console.log("Force: " + force);
console.log("Distance: " + distance);
console.log("Momentum: " + momentum);

console.log(); // Add some spacing

// Algorithm 6: Temperature Conversion with Multiple Formulas
let celsius = 25;
let fahrenheit = celsius * 9 / 5 + 32;
let kelvin = celsius + 273.15;
let celsius_from_f = (fahrenheit - 32) * 5 / 9;
let celsius_from_k = kelvin - 273.15;

console.log("Celsius: " + celsius);
console.log("Fahrenheit: " + fahrenheit);
console.log("Kelvin: " + kelvin);
console.log("F to C: " + celsius_from_f);
console.log("K to C: " + celsius_from_k);

console.log(); // Add some spacing

// Algorithm 7: Statistical Calculations
let num1 = 10;
let num2 = 20;
let num3 = 30;

let sum = num1 + num2 + num3;
let average = sum / 3;
let range = num3 - num1; // assuming num3 is largest, num1 is smallest
let sum_of_squares = num1 * num1 + num2 * num2 + num3 * num3;
let mean_of_squares = sum_of_squares / 3;
let variance = mean_of_squares - average * average;
let std_deviation = Math.sqrt(variance);

console.log("Sum: " + sum);
console.log("Average: " + average);
console.log("Range: " + range);
console.log("Variance: " + variance);
console.log("Standard Deviation: " + std_deviation.toFixed(2));

File: lua/stage-2/level-3/solution.lua
-- Algorithm 1: Complex Arithmetic Expression
local a = 10
local b = 5
local c = 2

local result = (a + b) * c - a / c
print("Result: " .. result)

local result2 = a + b * c - a / c
print("Result without parentheses: " .. result2)

local result3 = ((a + b) * c - a) / c
print("Result with different grouping: " .. result3)

print() -- Add some spacing

-- Algorithm 2: Quadratic Formula Calculator
local a2 = 1  -- Renamed to avoid conflict
local b2 = -5
local c2 = 6

local discriminant = b2 * b2 - 4 * a2 * c2
local sqrt_discriminant = math.sqrt(discriminant)
local root1 = (-b2 + sqrt_discriminant) / (2 * a2)
local root2 = (-b2 - sqrt_discriminant) / (2 * a2)

print("Root 1: " .. root1)
print("Root 2: " .. root2)

print() -- Add some spacing

-- Algorithm 3: Compound Interest Calculator
local principal = 1000
local rate = 0.05
local time = 10
local compounds_per_year = 12

local amount = principal * math.pow(1 + rate / compounds_per_year, compounds_per_year * time)
local interest = amount - principal

print("Principal: $" .. principal)
print("Final Amount: $" .. string.format("%.2f", amount))
print("Interest Earned: $" .. string.format("%.2f", interest))

print() -- Add some spacing

-- Algorithm 4: Geometric Calculations
local radius = 5
local side = 4
local base = 6
local height = 8

local circle_area = math.pi * radius * radius
local circle_circumference = 2 * math.pi * radius
local square_area = side * side
local square_perimeter = 4 * side
local triangle_area = 0.5 * base * height

print("Circle - Area: " .. string.format("%.2f", circle_area) .. ", Circumference: " .. string.format("%.2f", circle_circumference))
print("Square - Area: " .. square_area .. ", Perimeter: " .. square_perimeter)
print("Triangle - Area: " .. triangle_area)

print() -- Add some spacing

-- Algorithm 5: Physics Formula Calculator
local mass = 5.5
local velocity = 10
local acceleration = 9.8
local time2 = 3  -- Renamed to avoid conflict

local kinetic_energy = 0.5 * mass * velocity * velocity
local force = mass * acceleration
local distance = velocity * time2 + 0.5 * acceleration * time2 * time2
local momentum = mass * velocity

print("Kinetic Energy: " .. kinetic_energy)
print("Force: " .. force)
print("Distance: " .. distance)
print("Momentum: " .. momentum)

print() -- Add some spacing

-- Algorithm 6: Temperature Conversion with Multiple Formulas
local celsius = 25
local fahrenheit = celsius * 9 / 5 + 32
local kelvin = celsius + 273.15
local celsius_from_f = (fahrenheit - 32) * 5 / 9
local celsius_from_k = kelvin - 273.15

print("Celsius: " .. celsius)
print("Fahrenheit: " .. fahrenheit)
print("Kelvin: " .. kelvin)
print("F to C: " .. celsius_from_f)
print("K to C: " .. celsius_from_k)

print() -- Add some spacing

-- Algorithm 7: Statistical Calculations
local num1 = 10
local num2 = 20
local num3 = 30

local sum = num1 + num2 + num3
local average = sum / 3
local range = num3 - num1 -- assuming num3 is largest, num1 is smallest
local sum_of_squares = num1 * num1 + num2 * num2 + num3 * num3
local mean_of_squares = sum_of_squares / 3
local variance = mean_of_squares - average * average
local std_deviation = math.sqrt(variance)

print("Sum: " .. sum)
print("Average: " .. average)
print("Range: " .. range)
print("Variance: " .. variance)
print("Standard Deviation: " .. string.format("%.2f", std_deviation))